app/controllers/api/v1/users_controller.rb

```diff
@@ -9,8 +10,14 @@ def interests
   end
 
   def update_interests
+    # Rails treats all incoming params as strings (e.g. "true" instead of `true`), but we're trying
+    # to merge with a json hash.  Recall that json has typed booleans. Also, Rails 4.2 has a better
+    # way to do this, see http://stackoverflow.com/a/19423406
+    booleanized_params =
+      Hash[ params[:tws_interests].map {|k,v| [k, v == true || v == "true"]} ]
```

Comment:
this line is very confusing because i am not very smart and don't like reading.
can you add a comment about what it is doing. something like:

"turns string 'true' to bool `true`"